### ISSUE TITLE
output file naming collision

### ISSUE THREAD

**Issue by rlwimi** [dempseyatgithub/BuildSettingExtractor#7]
Let me first tell a story. When I was 7, I'd go to Wal-Mart and play The Legend of Zelda, a game like I'd never seen before, while my mom shopped. I saved up my allowance for an excruciatingly long time and finally walked out of Wal-Mart one day the proud owner of the golden cartridge. I turned on the game and named my character "LINK", because that's his name. Then everything went all screwy. None of the dungeons were where they were supposed to be. The instruction booklet didn't help. Not even the Zelda issue of Nintendo Power could help me. I eventually deleted the character out of frustration and made a new character, and eventually I name it "MIKE", and everything was back to normal. When I finished the game, I found out about the second quest, where all the positions are jumbled, and eventually I found out that naming your character "LINK" is a shortcut to the second quest.

Photo montage several decades to today. I'm converting my project "Beam" with this wonderful tool you've written. In it's Preferences, I change the Output Filenames > Project name: to "Beam", of course, because that's the project's name. The main Target is also named "Beam". Later, as I began removing build settings from the pbxproj, I realized BuildSettingsExtractor hadn't created a set of xcconfigs at the Project level. So I dove in to figure out the problem and eventually it hit me that I have this problem with naming all the things literally.

**In short**, for a Target named "Project" or in the case that the user changes the Output Filenames > Project name: such that it collides with a Target name–and I'd reckon a vast majority of projects have an identically-named Target–one or the other set of xcconfigs will silently clobber the previously written set. This will be especially confusing for people who are new to xcconfigs and their relation to the build system. Those of us with a literal naming complex could use all the help we can get, I mean, a notice that something has gone awry.

**Comment by dempseyatgithub** [comment]
First, thanks for taking the time to write up the issue, and also for the pull request.

I'm wondering if there might be a better experience than writing nothing and an error of 'please try again'.  Possibly using the default name of the Project if there is a conflict (and alerting the user at the end), and possibly using an alternate default name 'Common' if there is a conflict because the user has a target named 'Project' (and also alerting the user at the end).  Or possibly just appending something to the project-level files, like '-ProjectLevel' to avoid the file name conflict and an alert explaining what happened.

Will definitely think about the different approaches.

**Comment by rlwimi** [comment]
Sounds great. No worries if you want to close the PR. I usually name xcconfigs more explicitly using Target and Configuration terms themselves, since it can get hairy if you take subsequent steps to build further hierarchy after the extraction.

**Comment by dempseyatgithub** [comment]
This is fixed by commit 0283c5e7e16bf52f645ebe33e298920530fca0ff.

The code detects if the project name conflicts with a target name and adds "-Project-Settings" to the base name of the project config files. It also alerts the user that this was done.

Thank you for reporting this! I will be closing the pull request.
